packages/flutter/lib/src/cupertino/form_section.dart

```diff
@@ -0,0 +1,237 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+
+// Used for iOS "Inset Grouped" padding, determined from SwiftUI's Forms in
+// iOS 14.2 SDK.
+const EdgeInsetsDirectional _kDefaultInsetGroupedRowsPadding =
+    EdgeInsetsDirectional.fromSTEB(16.5, 0.0, 16.5, 16.5);
+
+// Used for iOS "Inset Grouped" border radius, estimated from SwiftUI's Forms in
+// iOS 14.2 SDK.
+// TODO(edrisian): This should be a rounded rectangle once that shape is added.
+const BorderRadius _kDefaultInsetGroupedBorderRadius =
+    BorderRadius.all(Radius.circular(10.0));
+
+// Used to differentiate the edge-to-edge section with the centered section.
+enum _CupertinoFormSectionType { base, insetGrouped }
+
+/// An iOS-style form section.
+///
+/// The base constructor for [CupertinoFormSection] constructs an
+/// edge-to-edge style section which includes an iOS-style header, footer, rows,
+/// the dividers between rows, and borders on top and bottom of the rows.
+///
+/// The [CupertinoFormSection.insetGrouped] constructor creates a round-edged and
+/// padded section that is commonly seen in notched-displays like iPhone X and
+/// beyond. Creates an iOS-style header, footer, rows, and the dividers
+/// between rows. Does not create borders on top and bottom of the rows.
+///
+/// The [header] parameter sets the form section header. The section header lies
+/// above the [children] rows.
+///
+/// The [children] parameter is required and sets the list of rows shown in
+/// the section. The [children] parameter takes a list, as opposed to a more
+/// efficient builder function that lazy builds, because forms are intended to
+/// be short in row count. It is recommended that only [CupertinoFormRow] and
+/// [CupertinoTextFormFieldRow] widgets be included in the [children] list in
+/// order to retain the iOS look.
+///
+/// The [margin] parameter sets the padding for the [Column] encapsulating
```

Comment:
re Column: in the public doc, mention things that are part of the public API. If they're not part of the public API, it's a private implementation detail. Don't mention too specifically how internal things are implemented. Otherwise, you're promoting it to a public interface (i.e. if the user knows how Columns work and code to take advantage of its idiosyncrasies and later you refactor the implementation internally, it's a breaking change).

Reply:
I'm struggling to name this properly. What should I call the column? If I say "Rows" then there might be confusion in whether the margin applies to individual rows or the container of all the rows.

Reply:
Just say `The [margin] parameter sets the spacing around the content area of the section encapsulating [children]`. I just meant don't reveal the fact that it's technically implemented using a `Column` specifically.